core/trino-main/src/main/java/io/trino/operator/DeduplicatingDirectExchangeBuffer.java

```diff
@@ -612,9 +609,10 @@ public OutputSource createOutputSource(Set<TaskId> selectedTasks)
             verify(sinkInstanceHandle != null, "sinkInstanceHandle is expected to be initialized");
 
             // Finish ExchangeSink and create ExchangeSource asynchronously to avoid blocking an ExchangeClient thread for potentially substantial amount of time
-            ListenableFuture<ExchangeSource> exchangeSourceFuture = FluentFuture.from(toListenableFuture(exchangeSink.finish()))
+            ListenableFuture<Void> sinkFinishFuture = toListenableFuture(exchangeSink.finish());
+            exchangeSink = null;
```

Comment:
setting it to null after 
```
...
                        exchange.sinkFinished(sinkInstanceHandle);
                        return toListenableFuture(exchange.getSourceHandles());
                    }, executor)
                    .transform(exchangeManager::createSource, executor);
exchangeSink = null;
```
should be fine. No need for extra `sinkFinishFuture`